tests/Doctrine/Tests/ORM/Functional/SecondLevelCacheOneToOneTest.php

```diff
@@ -187,4 +189,32 @@ public function testPutAndLoadOneToOneBidirectionalRelation()
         $this->assertEquals($queryCount, $this->getCurrentQueryCount());
     }
 
+    public function testPutAndLoadNonCacheableOneToOne()
+    {
+        $this->assertNull($this->cache->getEntityCacheRegion(Client::CLASSNAME));
+        $this->assertInstanceOf('Doctrine\ORM\Cache\Region', $this->cache->getEntityCacheRegion(Token::CLASSNAME));
+
+        $client = new Client('FabioBatSilva');
+        $token  = new Token('token-hash', $client);
+
+        $this->_em->persist($client);
+        $this->_em->persist($token);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $queryCount = $this->getCurrentQueryCount();
+
+        $this->assertTrue($this->cache->containsEntity(Token::CLASSNAME, $token->getToken()));
+        $this->assertFalse($this->cache->containsEntity(Client::CLASSNAME, $client->getId()));
+
+        $entity = $this->_em->find(Token::CLASSNAME, $token->getToken());
```

Comment:
Can you try to remove this line and make sure the test still passes? From your code, it seems that you are trying to extract the identifier from the association, but if the association is not cached, then this means the cache does not hold any identifier for the association, right?